File: src/arduinobot_cpp_examples/src/simple_service_server.cpp
#include <rclcpp/rclcpp.hpp>
#include <memory>
#include <arduinobot_msgs/srv/add_two_ints.hpp>

using namespace std::placeholders;

class SimpleServiceServer : public rclcpp::Node
{
    public:
    SimpleServiceServer() : Node("simple_service_server"){
        service_ = create_service<arduinobot_msgs::srv::AddTwoInts>("add_two_ints", std::bind(&SimpleServiceServer::serviceCallback, this, _1, _2));
        RCLCPP_INFO(get_logger(), "Service add_two_ints operational");
    }

    private:
    rclcpp::Service<arduinobot_msgs::srv::AddTwoInts>::SharedPtr service_;

    void serviceCallback(const std::shared_ptr<arduinobot_msgs::srv::AddTwoInts::Request> req,
                        const std::shared_ptr<arduinobot_msgs::srv::AddTwoInts::Response> res)
    {
        RCLCPP_INFO_STREAM(get_logger(), "Message received a: " << req->a << " b: " << req->b);
        res->sum =  req->a + req->b;
        RCLCPP_INFO_STREAM(get_logger(), "Response " << res->sum);
    }
};

int main(int argc, char* argv[]){
    rclcpp::init(argc, argv);
    auto simpleServiceServer = std::make_shared<SimpleServiceServer>();
    rclcpp::spin(simpleServiceServer);
    rclcpp::shutdown();
    return 0;

}

File: src/arduinobot_py_examples/arduinobot_py_examples/simple_service_server.py
import rclpy
from rclpy.node import Node
from arduinobot_msgs.srv import AddTwoInts

class SimpleServiceServer(Node):
    def __init__(self):
        super().__init__('simple_service_server')
        self.service_ = self.create_service(AddTwoInts, "add_two_ints", self.serviceCallback)
        self.get_logger().info("Service Add_Two_Ints oprational")

    def serviceCallback(self, req, res):
        self.get_logger().info(f"New message received { req.a}, {req.b} ")
        res.sum = req.a + req.b
        self.get_logger().info(f"Response { res.sum }")
        return res
    
def main():
    rclpy.init()
    simpleServiceServer = SimpleServiceServer()
    rclpy.spin(simpleServiceServer)
    simpleServiceServer.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
    